lmdeploy/serve/openai/protocol.py

```diff
@@ -139,6 +139,45 @@ class ChatCompletionRequest(BaseModel):
     seed: Optional[int] = None
 
 
+class ContentPartTextParam(BaseModel):
+    type: Literal['text']
+    text: str
+
+
+class ImageURL(BaseModel):
+    url: str
+
+
+class ContentPartImageParam(BaseModel):
+    type: Literal['image_url']
+    image_url: ImageURL
+
+
+ContentPartParam = Union[ContentPartTextParam, ContentPartImageParam]
+
+
+class AssistantMessage(BaseModel):
+    role: Literal['assistant']
+    content: Union[str, List[ContentPartTextParam]]
+
+
+class SystemMessage(BaseModel):
+    role: Literal['system']
+    content: Union[str, List[ContentPartTextParam]]
+
+
+class UserMessage(BaseModel):
+    role: Literal['user']
+    content: Union[str, List[ContentPartParam]]
+
+
+ChatCompletionMessage = Union[SystemMessage, UserMessage, AssistantMessage]
```

Comment:
I don't think this is the best way to validate them. There are also other roles like `tool`, and maybe more in the future.